File: server/Services/IServices/IProjectService.cs
using Data.Entities;
using Dtos.Project;
using Utils;

namespace Services;

public interface IProjectService
{
  Task<ServiceResponse<Guid>> Create(string uid, ProjectPostDto project);
  Task<ServiceResponse<List<ProjectGetDto>>> GetAll(string uid);
  Task<ServiceResponse<ProjectGetDto>> GetById(Guid id, string uid);
  Task<ServiceResponse<List<ProjectMembersGetDto>>> GetAllMembers(Guid projectId, string uid);
  Task<ServiceResponse<List<ProjectIssuesStatusCountGetDto>>> GetAllIssuesStatusCount(Guid projectId, string uid);
}

File: server/src/services/interfaces/IProjectService.ts
export default interface IProjectService {
  createProject(uid: string, project: object): Promise<void>;
  getAllProjects(uid: string): Promise<void>;
  getProjectById(uid: string): Promise<void>;
  getAllMembers(uid: string): Promise<void>;
  getAllIssuesCount(uid: string): Promise<void>;
}
